components/screens/ProductDetails.js

```diff
@@ -4,7 +4,36 @@ import { Card } from 'react-native-elements';
 import { TouchableOpacity } from 'react-native-gesture-handler';
 import NavBar from '../NavBar';
 
-export default function ProductDetails({ navigation }) {
+export default function ProductDetails({ screenProps, navigation }) {
+
+    const handleWishListPress = ()=>{
+        alert("Added to wish list");
+
+        const updatedList = screenProps.wishList;
+        updatedList.push({
+                        key: navigation.getParam('key'),
```

Comment:
Loop through the navigation outside the `updatedList` push.
And check that you are passing the real value not null or undefined.

Reply:
I surrounded the push with console.log and it seems to be passing the values.